STYLE_GUIDE.md

```diff
@@ -194,12 +218,112 @@ from composer.path.to.module import my_file as my_file
 ```
 
 
-## 6. Hparams and Config Classes
+## 7. Documentation
+
+Composer uses [Google Style Docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html).
+
+The following guidelines apply to documentation.
+1.  Each function that needs a docstring must have its input arguments and return statement (if not None) annotated.
+1.  Each argument annotation should include the type. If the argument has a default value, the type annotation should
+    specify "optional", and the docstring should say the default value. Some examples:
+
+    ```python
+
+    def foo(bar: int):
+        """Foo.
+
+        Args:
+            bar (int): Required bar.
+        """
+        ...
+
+    def foo2(bar: int = 42):
+        """Foo2.
+
+        Args:
+            bar (int, optional): Optional bar. (default: ``42``)
+        """
+        ...
+
+    def foo3(bar: Optional[int] = None):
+        """Foo3.
+
+        Args:
+            bar (int, optional): Optional bar. (default: ``None``)
+        """
+        ...
+
+    def foo4(bar: Union[int, str] = 42):
+        """Foo4.
+
+        Args:
+            bar (int | str, optional): Optional bar. (default: ``42``)
+        """
+        ...
+
+    def foo5(bar: int) -> int:
+        """Foo5.
+
+        Args:
+            bar (int): Required bar.
+
+        Returns:
+            int: Description of return statement.
+        """
+        ...
+
+    def foo6(bar: int) -> Tuple[int, str]:
+        """Foo6.
+
+        Args:
+            bar (int): Required bar.
+
+        Returns:
+            a (int): Returned value.
+            b (str): Returned value.
+        """
+        ...
+    ```
+
+1.  For examples in docstrings, use `.. doctest::` or
+    `.. testcode::` . See the [Sphinx Doctest Extension](https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html)
+    for all of the available directives. Do not use `.. code-block::` for Python examples, as they are untested.
+
+    Any test fixtures for doctests should go in [docs/source/doctest_fixtures.py](docs/source//doctest_fixtures.py)
+    or in a `.. testsetup::` block.
+
+    For example:
+    ```python
+    def my_function(x: Optional[torch.Tensor]) -> torch.Tensor:
+        """blah function
+
+        Args:
+            input (torch.Tensor, optional): Your guess.
```

Comment:
Can you do Optional args with no default value? maybe just remove the Optional here